gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -19,17 +19,51 @@ under the License.
 
 package gremlingo
 
-import "golang.org/x/text/language"
+import (
+	"crypto/tls"
+	"golang.org/x/text/language"
+	"net/http"
+)
+
+// AuthInfo is an option struct that allows authentication information to be specified.
+// Authentication can be provided via http.Header Header directly.
+// Basic authentication can also be used via the BasicAuthInfo function.
+type AuthInfo struct {
+	Header   http.Header
+	Username string
+	Password string
+}
+
+// getHeader provides a safe way to get a header from the AuthInfo even if it is nil.
+// This way we don't need any additional logic in the transport layer.
+func (authInfo *AuthInfo) getHeader() http.Header {
+	if authInfo == nil {
+		return nil
+	} else {
+		return authInfo.Header
+	}
+}
+
+// getUseBasicAuth provides a safe way to get a if basic auth info is available from the AuthInfo even if it is nil.
+// This way we don't need any additional logic in the transport layer.
+func (authInfo *AuthInfo) getUseBasicAuth() bool {
+	return authInfo != nil && authInfo.Username != "" && authInfo.Password != ""
+}
+
+// BasicAuthInfo provides a way to generate AuthInfo. Enter username and password and get the AuthInfo back.
+func BasicAuthInfo(username string, password string) *AuthInfo {
```

Comment:
do we need similar method for generating AuthInfo from header? like 
`func HeaderAuthInfo(header http.Header) *AuthInfo {`